nuxeo-features/nuxeo-platform-userworkspace/nuxeo-platform-userworkspace-core/src/test/java/org/nuxeo/ecm/platform/userworkspace/core/tests/TestUserWorkspace.java

```diff
@@ -359,12 +359,14 @@ public void testCanRetrieveUserWorkspaceWithTrashedDomain() {
                                         .collect(Collectors.toList());
         TrashService trashService = Framework.getService(TrashService.class);
         trashService.trashDocuments(docs);
+        coreFeature.waitForAsyncCompletion();
```

Comment:
Does it mean that trashDocuments turned to async, or tests was buggy ?

Reply:
It did turn async, children of documents trashed are handled by BAF now

Reply:
then org.nuxeo.ecm.core.api.trash.TrashService#trashDocuments java doc should be updated accordingly like (at least remove the "Placeless documents are deleted immediately.")